### PR TITLE
verify indexes queries dont need schemalk

### PR DESCRIPTION
**Pull request by ldennis**

### CODE REVIEW

**Comment by adizaimi** [comment]
Looks good to me. Did master fail the test you are adding here?

**Comment by ldennis** [comment]
Yes, after adding the same sleep to master.

File: db/sqlinterfaces.c [hunk 1]
```diff
@@ -5740,11 +5740,16 @@ static int execute_verify_indexes(struct sqlthdstate *thd,
 {
     int rc;
     if (thd->sqldb == NULL) {
-        rdlock_schema_lk();
-        rc = sqlengine_prepare_engine(thd, clnt, 1);
-        unlock_schema_lk();
-        if (rc) {
-            return rc;
+        /* open sqlite db without copying rootpages */
+        rc = sqlite3_open_serial("db", &thd->sqldb, thd);
```
Comment (akshatsikarwar) [inline db/sqlinterfaces.c]: Won't this fail if a schemachange is updating schema variables which is used to bootstrap sqlite vm during open?

Comment (ldennis) [reply db/sqlinterfaces.c]: I guess only copying rootpages needs schemalk.

Comment (akshatsikarwar) [reply db/sqlinterfaces.c]: Specifically, code in create_sqlmaster_record. You can latch on to one of those pointers which can get freed by the time you deref.

Comment (ldennis) [reply db/sqlinterfaces.c]: I dont think `sqlite3_open_serial` derefs sqlmaster, no?

Comment (akshatsikarwar) [reply db/sqlinterfaces.c]: umm - it should.

Comment (akshatsikarwar) [reply db/sqlinterfaces.c]: `select * from sqlite_master` gets it from there (or cached rootpage info). Either way lock is required during copy.

Comment (ldennis) [reply db/sqlinterfaces.c]: Yes, `select * from sqlite_master` needs to load schema from rootpage. However, schema is not loaded for `select 1` or CTEs. I dont copy rootpage here so I dont think schemalk is required.

Comment (akshatsikarwar) [reply db/sqlinterfaces.c]: hmm - ok. This is fine then!

Comment (ldennis) [reply db/sqlinterfaces.c]: I have concern on stack if we call vm inline given that we are already deep in the stack in blockprocessor. And other than bp threads, convert record thd in sc_records.c also needs the vm. So I would keep the dispatch sql code for now. I can time the performance diff later.

**Pull request merged**